src/components/timecard/time-input-errors/TimeInputErrors.jsx

```diff
@@ -0,0 +1,118 @@
+import { PropTypes } from 'prop-types';
+import { useApplicationContext } from '../../../context/ApplicationContext';
+import {
+  formatLongDate,
+  formatTime,
+  getTimePeriodTypesMap,
+} from '../../../utils/time-entry-utils/timeEntryUtils';
+
+const TimeInputErrors = ({ clashingProperty, clashes }) => {
+  const { timePeriodTypes } = useApplicationContext();
+  const timePeriodTypesMap = getTimePeriodTypesMap(timePeriodTypes);
+
+  const displayClashingProperty = () => {
+    if (clashingProperty === 'startTime') {
+      return <p>Your start time must not overlap with another time period</p>;
+    }
+
+    if (clashingProperty === 'endTime') {
+      return <p>Your finish time must not overlap with another time period</p>;
+    }
+
+    return (
+      <p>
+        Your start and finish times must not overlap with another time period
+      </p>
+    );
+  };
+
+  const displaySingleTimeClash = () => {
+    const clash = clashes[0];
```

Comment:
nit: optional, could throw an error with a nice message before this if there are no clashes. In the rare circumstance that this component gets called an empty clashes array.

Reply:
👍 done

Reply:
nice. just a hint (no need to change), `==` would have been sufficient on ln 32 as array.length always returns an integer.